plan/plan.go

```diff
@@ -271,13 +271,11 @@ type Rel interface {
 	// result should be 3 columns consisting of the 5th, 2nd and 1st
 	// output columns from the underlying relation.
 	OutputMapping() []int32
-	// Remap utilizes the OutputMapping to construct the expected result
-	// record type from the output RecordType of the underlying Relation.
-	// If the output mapping is nil, then this just returns the input
-	// type.
-	Remap(types.StructType) types.StructType
-	// RecordType returns the output record type of the underlying relation
-	// as a struct type.
+	// directOutputSchema returns the output record type of the underlying
+	// relation as a struct type.  Mapping is not applied.
+	directOutputSchema() types.StructType
+	// RecordType returns the types used by all columns returned by
+	// this relation after applying any provided mapping.
 	RecordType() types.StructType
```

Comment:
can we redefine this so it returns a RecordType as opposed to StructType. StructType isn't actually right since relations don't return structs, they return sets of records. As a concrete distinction, StructType can be nullable and type variations. Neither of those things are relevant to a record type.